src/server/storage/storageAPI.js

```diff
@@ -42,20 +44,63 @@ const setup = (app: Router, storage: StorageAPI) => {
         identifier: userRecord.loggedInAs,
         createdDate: new Date().toString()
       })
+
+      const mauticRecordPromise =
+        process.env.NODE_ENV === 'development'
+          ? Promise.resolve({})
+          : Mautic.createContact(user).catch(e => {
+              log.error('Create Mautic Record Failed', e)
+            })
+
+      const secureHash = md5(`${user.email}${conf.secure_key}`)
+      const web3RecordPromise = user.w3Token
+        ? Promise.rewolve()
+        : fetch(`${conf.web3SiteUrl}/api/wl/user`, {
```

Comment:
now that we do this for all users, what's going to happen for users that came from W3? ie email is already on W3?

Reply:
There is a condition for web3RecordPromise to not do this for users that came with web3 token

Reply:
when is this w3Token saved to the user's profile?
when is it saved when a user came from W3?

Reply:
It is saved when the user came from web3 with token

Reply:
iI don't think so. I think you are mixing up the user profile on the dapp and the user record on the server, they are not the same.

Reply:
The web3 token saved to server user record while there is /user/add request initiated. So the web3 token present in that request payload